app/controllers/publishers/site_banners_controller.rb

```diff
@@ -5,14 +5,104 @@ class Publishers::SiteBannersController < ApplicationController
 
   MAX_IMAGE_SIZE = 10_000_000
 
-  def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
+  #Fetch Banner by channel_id, if it does not exist create a template banner.
+  def fetch
+    site_banner = current_publisher.site_banners.where(channel_id: params[:channel_id]).first
+    if site_banner.present?
+      data = site_banner.read_only_react_property
+      data[:backgroundImage] = data[:backgroundUrl]
+      data[:logoImage] = data[:logoUrl]
+      render(json: data.to_json)
+    else
+      site_banner = SiteBanner.new
+      site_banner.update(
+        publisher_id: current_publisher.id,
+        channel_id: params[:channel_id],
+        title: 'Brave Rewards',
+        description:
+'Thanks for stopping by. We joined Brave\'s vision of protecting your privacy because we believe that fans like you would support us in our effort to keep the web a clean and safe place to be.
+
+Your tip is much appreciated and it encourages us to continue to improve our content.',
+        donation_amounts: [1, 5, 10],
+        default_donation: 5,
+        social_links: {'youtube': '', 'twitter': '', 'twitch': ''}
```

Comment:
Although I prefer single quotes our linter is configured to do double quotes (even if it doesn't run on this file yet) we should probably be consistent.

@yachtcaptain23 what are your thoughts?

Reply:
best practice is to use `"` for ruby for this project

Reply:
Done ✔️